wave_optimization/include/wave/optimization/factor_graph/variable.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+#define WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+
+struct Variable {
+    size_t id;
+    size_t size;
+    VecX data;
+
+    Variable() : id{0}, size{0}, data{0} {}
+    Variable(size_t id, size_t size) : id{id}, size{size}, data{size} {}
+
+    std::string toString() {
+        std::ostringstream oss;
+
+        oss << "[";
+        oss << "id: " << this->id << ", ";
+        oss << "size: " << this->size;
+        oss << "]";
+
+        return oss.str();
+    }
+
+    void print() {
+        std::cout << this->toString() << std::endl;
+    }
+
+    bool operator<(const Variable &other) const {
+        if (this->id < other.id) {
+            return true;
+        }
+        return false;
+    }
+
+    VecX &getData() {
+        return data;
+    }
+};
+
+struct PoseVar : Variable {
+    PoseVar(size_t id) : Variable{id, 6} {
```

Comment:
You can certainly use it in Ceres optimization, not sure if it's a simple as "grab the address".

Keep it this way for now, refactor later?